planet/specs.py

```diff
@@ -111,15 +111,18 @@ def get_item_types(product_bundle=None):
     '''If given product bundle, get specific item types supported by Orders
     API. Otherwise, get all item types supported by Orders API.'''
     spec = _get_product_bundle_spec()
-    if product_bundle:
-        item_types = spec['bundles'][product_bundle]['assets'].keys()
-    else:
-        product_bundle = get_product_bundles()
-        all_item_types = []
-        for bundle in product_bundle:
-            all_item_types += [*spec['bundles'][bundle]['assets'].keys()]
-        item_types = set(all_item_types)
-    return item_types
+    try:
+        if product_bundle:
+            item_types = spec['bundles'][product_bundle]['assets'].keys()
+        else:
+            product_bundle = get_product_bundles()
+            all_item_types = []
+            for bundle in product_bundle:
+                all_item_types += [*spec['bundles'][bundle]['assets'].keys()]
+            item_types = set(all_item_types)
+        return item_types
+    except KeyError:
+        raise SpecificationException()
```

Comment:
Add a specific message?

Reply:
I think the `SpecificationException` raised by `_validate_field()`, which `validate_bundle()` calls is perfect for this case. I'm not quite sure if this is fine, but could I just raise `validate_bundle(product_bundle)`, as that'll raise a `SpecificationException` with the proper message?

Reply:
See https://github.com/planetlabs/planet-client-python/pull/677/commits/614ebe5d0d7edd7d34b052ec81e266d757cbdacc#diff-0c5c73e85ab9430a67e6b85ba97b8d4fd186f5240bc8f4c1ddb7ef52fade26feR123

Reply:
@kevinlacaille it occurs to me that with some small changes to SpecificationException, it could be raised from anywhere and you wouldn't need to call any one particular function to trigger it. For example.

```
class SpecificationException(Exception):
    def __init__(value, supported, field_name):
        self.value = value
        self.supported = supported
        self.field_name = field_name
    def __str__(self):
        return f'{self.field_name} - \'{self.value}\' is not one of {self.opts}.'
```

Reply:
Oh, I really prefer this - great suggestion!